src/trace-writer.ts

```diff
@@ -15,8 +15,10 @@
  */
 
 import * as common from '@google-cloud/common';
+import {AxiosError} from 'axios';
```

Comment:
Ugh.. This is an abstraction leak from gcp-metadata, by bypassing axios errors to the client code. /cc @JustinBeckwith

Reply:
Yeah, little bit.  What would you suggest?

Reply:
An obvious answer would be "catch errors at the API boundaries and transform them into gcp-metadata's own errors". But I'm not sure how simple it'd be to do that.

I think at least we should document what kind of fields are available in the error thrown by gcp-metadata, and make it not be broken when we switch axios to something else, if ever, later.